config/routes.rb

```diff
@@ -43,6 +43,10 @@
       end
 
       resources :search, only: :index
+
+      resources :subscriptions, only: %i[create]
+      delete '/unsubscription', to: 'subscriptions#destroy'
+      get '/subscription', to: 'subscriptions#show'
     end
```

Comment:
I may have misunderstood your solution a bit, but perhaps such an option may exist.

```
resources :subscriptions, only: [:create, :show, :destroy] do
  delete 'unsubscription', on: :member, to: 'subscriptions#destroy'
  get 'subscription', on: :member, to: 'subscriptions#show'
end

```
but I'm not sure

Reply:
:member used to identify route with :id throw a subscription model this solution made rails routes like that

DELETE /api/v1/subscriptions/:id/unsubscription
GET    /api/v1/subscriptions/:id/subscription

if you're looking for a subscribe service I didn't use any :id for GET or DELETE only email